toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -163,6 +163,9 @@ func addNodeToGraphHelper(g *PkgGraph, node *PkgNode) (newNode *PkgNode, err err
 		node.Architecture,
 		node.SourceRepo,
 	)
+
+	// Updating node's ID for the sake of equality testing.
+	node.nodeID = newNode.nodeID
```

Comment:
I'm a bit leery of this, generally moving IDs around isn't going to end well. I've also added an equality test here, maybe it helps? https://github.com/microsoft/CBL-Mariner/pull/6368/files#diff-57fa73a04ce75440c4e9f3d332d831a6d03fea3e663b4f3d00e656a396b167c6 (Ignore the N+1 changes in that PR, just look at the last commits)

Reply:
Sounds good, @dmcilvaney. I'll "borrow" your equality test then.;)